Sources/Amplitude/Amplitude.m

```diff
@@ -488,14 +488,16 @@ - (void) checkInForeground {
         if (app != nil) {
             UIApplicationState state = app.applicationState;
             if (state != UIApplicationStateBackground) {
+                self->_inForeground = YES;
                 [self runOnBackgroundQueue:^{
+    #else
+                    self->_inForeground = YES;
```

Comment:
Is this needed? And do we want the else break here?

Reply:
Before my changes `self->_inForeground = YES;` was executed in any case.
Now if `!TARGET_OS_OSX && !TARGET_OS_WATCH` is true it is executed in main thread before `self runOnBackgroundQueue`, otherwise - it is executed before `refreshDynamicConfig`. So `self->_inForeground = YES` is still executed in any case (now in slightly different locations).
Does it look correct?

Reply:
Taking another look, it seems fine.